app/src/chainlit.py

```diff
@@ -253,65 +247,27 @@ def _get_retrieval_metadata(result: OnMessageResult) -> dict:
     }
 
 
-@cl.action_callback("download_batch_results")
-async def on_download(action: Action) -> None:
-    """Handle downloading of completed batch results"""
-    file_path = BATCH_RESULTS.get(action.value)
-    if file_path and os.path.exists(file_path):
+async def _batch_proccessing(file: AskFileResponse) -> None:
+    await cl.Message(
+        author="backend",
+        content="Received file, processing...",
+    ).send()
+
+    try:
+        engine: chat_engine.ChatEngineInterface = cl.user_session.get("chat_engine")
+        result_file_path = await batch_process(file.path, engine)
+
+        # E.g., "abcd.csv" to "abcd_results.csv"
+        result_file_name = file.name.removesuffix(".csv") + "_results.csv"
+
         await cl.Message(
-            content="Here are your processed results:",
-            elements=[cl.File(name=os.path.basename(file_path), path=file_path)],
+            content="File processed, results attached.",
+            elements=[cl.File(name=result_file_name, path=result_file_path)],
         ).send()
-        # Optionally cleanup
-        BATCH_RESULTS.pop(action.value)
-    else:
-        await cl.Message(content="Results no longer available").send()
 
-
-async def _batch_proccessing(file: AskFileResponse) -> None:
-    try:
-        engine: ChatEngineInterface = cl.user_session.get("chat_engine")
-
-        # Create initial progress message
-        progress_msg = Message(content="Starting batch processing...")
-        await progress_msg.send()
-
-        # Process in background task
-        task_id = f"batch_{file.name}_{id(file)}"
-
-        async def process_and_notify() -> None:
-            try:
-
-                async def update_progress(current: int, total: int) -> None:
-                    # Create new message with updated content
-                    progress_msg.content = f"Processing {current}/{total} questions..."
-                    await progress_msg.update()
```

Comment:
@fg-nava Feel free to add `progress_callback=update_progress` back in once we're confident BP works for the users.